From 222c61177367ea2ce07beec8e4ff04e8a1dacda7 Mon Sep 17 00:00:00 2001
From: Samuel Ortiz <sameo@linux.intel.com>
Date: Fri, 25 Jul 2014 15:15:56 +0200
Subject: [PATCH 1/2] rfkill: Support BCM4752E GPS device

Some devices export the BCM4752 GPS device through ACPI tables with
an "BCM4752E" _HID to distinguish BCM4752 from BCM4752 + external
LNA setups.
In order to control both setups power paths from rfkill, we need
to add "BCM4752E" to the list of supported ACPI devices from
rfkill-gpio.c.

Change-Id: Ie0cc7f198a10981653e5f451d4ec2835a7aeaaed
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
---
 drivers/acpi/acpi_platform.c | 1 +
 net/rfkill/rfkill-gpio.c     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/acpi/acpi_platform.c b/drivers/acpi/acpi_platform.c
index f73a8a0..4910936 100644
--- a/drivers/acpi/acpi_platform.c
+++ b/drivers/acpi/acpi_platform.c
@@ -38,6 +38,7 @@ static const struct acpi_device_id acpi_platform_device_ids[] = {
 	{ "BCM2E3D" },
 	{ "RTL8723" },
 	{ "OBDA8723" },
+	{ "BCM4752E" },
 
 	/* Intel Smart Sound Technology */
 	{ "INT33C8" },
diff --git a/net/rfkill/rfkill-gpio.c b/net/rfkill/rfkill-gpio.c
index 73299df..e7e9a95 100644
--- a/net/rfkill/rfkill-gpio.c
+++ b/net/rfkill/rfkill-gpio.c
@@ -172,6 +172,7 @@ static const struct acpi_device_id rfkill_acpi_match[] = {
 	{ "OBDA8723", RFKILL_TYPE_BLUETOOTH },
 	{ "BCM4752", RFKILL_TYPE_GPS },
 	{ "LNV4752", RFKILL_TYPE_GPS },
+	{ "BCM4752E", RFKILL_TYPE_GPS },
 	{ },
 };
 MODULE_DEVICE_TABLE(acpi, rfkill_acpi_match);
-- 
1.8.3.2

